SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,256 @@
+#include "../../../../Common/include/geometry/CGeometry.hpp"
+#include "../../../../Common/include/CConfig.hpp"
+#include "../../fluid/CFluidModel.hpp"
+
+using namespace std;
+
+class CTurbomachineryPrimitiveState {
+  private:
+  su2double Density, Pressure, TangVelocity;
+  vector <su2double> Velocity;
+  public:
+  CTurbomachineryPrimitiveState();
+
+  CTurbomachineryPrimitiveState(vector<su2double> TurboPrimitives,
+                                unsigned short nDim,
+                                su2double tangVel);
+
+  su2double GetDensity() const { return Density; }
+
+  su2double GetPressure() const { return Pressure; }
+
+  su2double GetTangVelocity() const { return TangVelocity; }
+
+  vector <su2double> GetVelocity() const { return Velocity; }
+
+};
+
+class CTurbomachineryCombinedPrimitiveStates {
+  private:
+  CTurbomachineryPrimitiveState InletPrimitiveState;
+  CTurbomachineryPrimitiveState OutletPrimitiveState;
+  public:
+
+  CTurbomachineryCombinedPrimitiveStates(const CTurbomachineryPrimitiveState &inletPrimitiveState,
+                                         const CTurbomachineryPrimitiveState &outletPrimitiveState);
+
+  CTurbomachineryPrimitiveState GetInletPrimitiveState() const { return InletPrimitiveState; }
+
+  CTurbomachineryPrimitiveState GetOutletPrimitiveState() const { return OutletPrimitiveState; }
+};
+
+class CTurbomachineryState {
+  private:
+  su2double Density, Pressure, Entropy, Enthalpy, Temperature, TotalTemperature, TotalPressure, TotalEnthalpy;
+  su2double AbsFlowAngle, FlowAngle, MassFlow, Rothalpy, TotalRelPressure;
+  vector <su2double> Velocity, RelVelocity, Mach, RelMach;
+  su2double Area, Radius;
+
+
+  public:
+  CTurbomachineryState();
+
+  CTurbomachineryState(unsigned short nDim, su2double area, su2double radius);
+
+  void ComputeState(CFluidModel& fluidModel, const CTurbomachineryPrimitiveState &primitiveState);
+
+  su2double GetDensity() const { return Density; }
+
+  su2double GetPressure() const { return Pressure; }
+
+  su2double GetEntropy() const { return Entropy; }
+
+  su2double GetEnthalpy() const { return Enthalpy; }
+
+  su2double GetTemperature() const { return Temperature; }
+
+  su2double GetTotalTemperature() const { return TotalTemperature; }
+
+  su2double GetTotalPressure() const { return TotalPressure; }
+
+  su2double GetTotalRelPressure() const { return TotalRelPressure; }
+
+  su2double GetTotalEnthalpy() const { return TotalEnthalpy; }
+
+  su2double GetAbsFlowAngle() const { return AbsFlowAngle; }
+
+  su2double GetFlowAngle() const { return FlowAngle; }
+
+  su2double GetMassFlow() const { return MassFlow; }
+
+  su2double GetRothalpy() const { return Rothalpy; }
+
+  vector <su2double> GetVelocity() const { return Velocity; }
+
+  vector <su2double> GetMach() const { return Mach; }
+
+  su2double GetVelocityValue() const {
+    return sqrt(inner_product(Velocity.begin(), Velocity.end(), Velocity.begin(), 0.0));
```

Comment:
is this boost? We have a geometry_toolbox with norms, dot products, etc. for when you replace boost.